packages/graphback-runtime-mongodb/src/OffixMongoDBDataProvider.ts

```diff
@@ -10,17 +10,50 @@ import { MongoDBDataProvider } from './MongoDBDataProvider';
  * https://offix.dev/docs/conflict-server#structure-of-the-conflict-error
  */
 export class OffixMongoDBDataProvider<Type = any, GraphbackContext = any> extends MongoDBDataProvider<Type, GraphbackContext> {
+  protected updatedAtField: string;
+  protected createdAtField: string;
 
   public constructor(baseType: GraphQLObjectType, client: any) {
     super(baseType, client);
+    this.createdAtField = undefined;
+    this.updatedAtField = undefined;
+    Object.keys(baseType.getFields()).forEach((k: string) => {
+      if (baseType.getFields()[k]?.astNode?.directives) {
+        baseType.getFields()[k].astNode.directives.forEach((directive: any) => {
+          if (directive?.name?.value === "createdAt") {
+            if (this.createdAtField === undefined) {
+              this.createdAtField = baseType.getFields()[k]?.name;
```

Comment:
This is an idea to make more generic directives, it would mean they would need to be implemented per database type, but means logic could be reused in any CRUD method

```ts
this.directives: DirectiveObj = {
  updatedAt: {
    fieldName: baseType.getFields()[k]?.name,
    resolve: (id) => {
      return (new ObjectId(id)).getTimestamp()
    }
  }
}
```

Then in your CRUD function:

```ts
const { updatedAt } = this.directives;

res = await super.update({
  ...res,
  id: res.id,
  [updatedAt.fieldName]: updatedAt.resolve(res.id)
});
```

Reply:
Sorry! I am not sure I understand :(

What I was thinking was we would store both of these things in the database itself so that would eliminate the need to doing all the hacking that is present in this PR currently

Reply:
What you are saying is a good idea, I was also thinking it would be better to unify these 'things'(directives/metadata?), but I am not sure I completely understand your example

Reply:
> What I was thinking was we would store both of these things in the database itself

Store what exactly in the database itself?

> Sorry! I am not sure I understand :(

No problem :)

Basically my example is very similar to yours, except instead of implementing the logic in the provider method, it could be be built beforehand, Ive made a little gist prototype:

https://gist.github.com/craicoverflow/a165dbb42688e26d794f6f0a18bd2ebc#file-offixmongodbdataproviderwithdirectives-ts-L21

https://gist.github.com/craicoverflow/a165dbb42688e26d794f6f0a18bd2ebc#file-offixmongodbdataproviderwithdirectives-ts-L86

Reply:
Oh I see! That is indeed useful!

> Store what exactly in the database itself?

I was actually thinking something else, so that statement isn't actually relevant to the discussion here